### PR TITLE
Tax and some other minor features

### PR DESCRIPTION
**Pull request by gamingdiamond982**
Added all of the stuff needed for tax, and some other features requested by imade, some other changes were made by my ide, stuff like extra whitespace etc. was considering moving tax to another file but I saw you had put all your objects in here so decided to follow suit

### CODE REVIEW

**Comment by jonathanvdc** [comment]
I think this PR reverses the fractional currency feature I just added...

EDIT: fixed

File: src/accounting.py [hunk 1]
```diff
@@ -327,6 +330,13 @@ def open_account(self, id: AccountId, account_uuid=None):
         self.inv_accounts[account].append(id)
         return account
 
+    def delete_account(self, id: AccountId, account_uuid=None):
```
Comment (jonathanvdc) [inline src/accounting.py:333]: Nice addition. But what about recurring transfers? Don't we need to check that there are none of those left before deleting?

File: src/accounting.py [hunk 2]
```diff
@@ -383,6 +393,9 @@ def print_money(self, author: AccountId, account: Account, amount: Fraction):
         """Prints `amount` of money on the authority of `author` and deposits it in `account`."""
         account.balance += amount
 
+    def remove_funds(self, author: Account, account: Account, amount: Fraction):
```
Comment (jonathanvdc) [inline src/accounting.py:405]: Isn't this just money printing with negative amounts?

File: src/accounting.py [hunk 3]
```diff
@@ -585,13 +603,127 @@ def create_initial_ledger(unannotated_ledger_path, result_path, leading_zero_cou
     with open(result_path, 'w') as f:
         f.writelines(line + '\n' for line in lines)
 
+
+class WealthTaxBracket:
```
Comment (jonathanvdc) [inline src/accounting.py:616]: Nice data structure.

File: src/accounting.py [hunk 4]
```diff
@@ -585,13 +603,127 @@ def create_initial_ledger(unannotated_ledger_path, result_path, leading_zero_cou
...
+
+    def get_end(self):
+        return self.end
+
+    def get_tax(self, account):
+        bal = account.get_balance()
+        if bal < self.start:
+            return 0
+        elif self.end is None or bal <= self.end:
+            tax_amount = round(((bal - self.start) / 100) * self.tax_rate)
```
Comment (jonathanvdc) [inline src/accounting.py:657]: Do we still need to round now that we have fractions?

File: src/accounting.py [hunk 5]
```diff
@@ -585,13 +603,127 @@ def create_initial_ledger(unannotated_ledger_path, result_path, leading_zero_cou
...
+
+
+
+class TaxException(Exception):
+    pass
+
+
+class TaxMan:
+    def __init__(self, server, tax_regularity=28, auto_tax=False):
+        self.tax_brackets = {}
```
Comment (jonathanvdc) [inline src/accounting.py:671]: Multiple tax brackets? Nice.

File: src/accounting.py [hunk 6]
```diff
@@ -694,13 +834,24 @@ def _read_ledger(self, ledger_path):
...
+            elif cmd == 'force-tax':
+                pass
             else:
                 raise Exception("Unknown ledger command '%s'." % cmd)
 
     def _ledger_write(self, *args, t=None):
         if t is None:
             t = time.time()
-        elems = [str(t)] + ['%d/%d' % (x.numerator, x.denominator) if isinstance(x, Fraction) else str(x) for x in args]
+        elems = [str(t)] + list(map(str, args))
```
Comment (jonathanvdc) [inline src/accounting.py]: Please restore my edit.

**Review by jonathanvdc — COMMENTED** [review]
Looks good. I just have a few remarks. Can you address them?

File: src/accounting.py (hunk 2 above)
Comment (gamingdiamond982) [reply src/accounting.py:405]: yes it was one of the features imade requested

File: src/accounting.py (hunk 6 above)
Comment (gamingdiamond982) [reply src/accounting.py]: sorry about that will fix

File: src/accounting.py (hunk 1 above)
Comment (gamingdiamond982) [reply src/accounting.py:333]: ah we do, I forgot about that and it didn't come up in testing
Edit: Fixed

File: src/accounting.py (hunk 4 above)
Comment (gamingdiamond982) [reply src/accounting.py:657]: no we don't when I wrote it we did tho

File: src/accounting.py (hunk 6 above)
Comment (gamingdiamond982) [reply src/accounting.py]: fixed

File: src/accounting.py (hunk 4 above)
Comment (jonathanvdc) [reply src/accounting.py:657]: Can you get rid of the rounding then?

Comment (gamingdiamond982) [reply src/accounting.py:657]: actually in simdem law it legally has to be rounded to the nearest whole number

Comment (jonathanvdc) [reply src/accounting.py:657]: smh that's a misfeature but okay

**Comment by jonathanvdc** [comment]
Okay let's do this.

**Pull request merged**